clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java

```diff
@@ -301,6 +301,24 @@ public void testInvalidKeystorePassword() throws Exception {
             // Expected exception
         }
     }
+
+    /**
+     * Tests that client connections can be created to a server
+     * if null truststore password is used
+     */
+    @Test
+    public void testNullTruststorePassword() throws Exception {
```

Comment:
We pass the configuration when creating the server via `NetworkTestUtils.createEchoServer`. This eventually calls `SslFactory.configure` which eventually calls `SslFactory.createTruststore`. 

If the password passed is not set `null` is used and integrity checking is disabled. 

If any password is passed, even one with a typo, integrity checking is enabled and it must match the password of the trust store. This ensures no one can change the truststore without knowing and setting the expected password.